src/Sidebar/Sidebar.tsx

```diff
@@ -8,12 +7,37 @@ import {
   SidebarHeaderStyles,
 } from "./sidebar.style";
 
-const renderableChildrenPropType = PropTypes.oneOfType([
-  PropTypes.arrayOf(PropTypes.node),
-  PropTypes.node,
-]);
+type LinkProps = {
+  iconName: string;
+  active?: boolean;
+  href: string;
+};
+
+type HeaderProps = {
+  logoUrl: string;
+  logoAltText: string;
+  href: string;
+};
+
+interface LinkFC<T> extends React.FC<T> {
```

Comment:
This is a bit over complicated. Display name is a react functional component feature, and so it exists on the FC type already.
Secondly, no need to use generics or extend the FC type here, you can use an intersection, and replace all this code with: 

```typescript
const Link: React.FC<LinkProps> = ({ 
  // .. 
})
Link.displayName = "Sidebar.Link";

const Header: React.FC<HeaderProps> = ({ 
  // .. 
})
Header.displayName = "Sidebar.Header";

interface SideBarSubComponents {
  Link: React.FC<LinkProps>;
  Header: React.FC<HeaderProps>;
}

const Sidebar: React.FC & SideBarSubComponents = ({ children }) => {
};

Sidebar.Header = Header;
Sidebar.Link = Link;
export default Sidebar;
```